types/src/reconfiguration_event_subscription.rs

```diff
@@ -0,0 +1,9 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+/// Trait that must be implemented by a subscription to reconfiguration events emitted
+/// by state sync
+
+pub trait ReconfigurationEventSubscription: Send + Sync {
+    fn publish(&self, payload: usize);
```

Comment:
does this need to be `Sync`? it looks like it only needs `Send`. If you make `publish` take a `&mut self` instead, we can also send on internal channels without extra clones.

Reply:
Just tried removing Sync, and looks like we need it because it gets shared between threads in state sync